### ISSUE TITLE
Expand output paths for `flutter build ios-framework`, allow ~

### ISSUE THREAD

**Issue by jmagman** [flutter/flutter#45738]
## Steps to Reproduce

```
flutter build ios-framework --output=~/Desktop
```

Creates a `~` directory in the working directory instead of expanding ~

**Comment by xster** [comment]
doh

**Comment by jmagman** [comment]
https://github.com/dart-lang/sdk/issues/18466

**Comment by xster** [comment]
lol I had the same reaction: umm this should be a one line library use fix... oh, package:file doesn't do this...

**Comment by xster** [comment]
@jmagman I think you fixed this right?

**Comment by jmagman** [comment]
> @jmagman I think you fixed this right?

No, there's no dart support for expanding the tilde, so it would require... ffi?
We could print a warning if we parse a ~ though.
